librz/util/sys.c

```diff
@@ -1456,9 +1456,15 @@ RZ_API int rz_sys_pipe_close(int fd) {
 // Use this lock to wraps pipe, close, exec*, system to ensure all pipe file
 // descriptors are either created AND set as CLOEXEC or not created at all.
 static RzThreadLock *sys_pipe_mutex;
+bool is_child = false;
+
+static void set_child(void) {
+	is_child = true;
+}
 
 __attribute__ ((constructor)) static void sys_pipe_constructor(void) {
 	sys_pipe_mutex = rz_th_lock_new (true);
+	pthread_atfork (NULL, NULL, set_child);
```

Comment:
Does this still work in threads that are not the main thread? From the man page:
```
The pthread_atfork() function registers fork handlers that are to be executed when fork(2) is called by this thread.
```

Reply:
Mmmm good point... I need to investigate, it seems it doesn't from what you say.

Reply:
The "solution", since we have a `rz_sys_fork` anyway, could be to just implement the `pthread_atfork` logic in `rz_sys_fork`. That should work.